src/NServiceBus.AzureStoragePersistence/SagaPersisters/Azure/AzureSagaPersister.cs

```diff
@@ -41,18 +41,26 @@ public AzureSagaPersister(string connectionString, bool autoUpdateSchema)
             secondaryIndeces = new SecondaryIndexPersister(GetTable, ScanForSaga, Persist);
         }
 
-        DictionaryTableEntity GetDictionaryTableEntity(string sagaId, Type entityType)
+        async Task<DictionaryTableEntity> GetDictionaryTableEntity(string sagaId, Type entityType)
         {
             var tableName = entityType.Name;
             var table = client.GetTableReference(tableName);
 
             var query = new TableQuery<DictionaryTableEntity>().Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, sagaId));
 
-            var tableEntity = table.ExecuteQuery(query).SafeFirstOrDefault();
-            return tableEntity;
+            try
+            {
+                var tableEntity = (await table.ExecuteQueryAsync(query).ConfigureAwait(false)).SafeFirstOrDefault();
+                return tableEntity;
+            }
+            catch (StorageException)
+            {
+                // Not found
+                return null;
```

Comment:
I think it might be not true. If the `StorageException` happened because of the `503` on the server side like in [this example] (http://stackoverflow.com/questions/29727269/azure-webjob-stopped-with-unhandled-exception-503-server-unavailable) this would result in treating the entity as non-existent, when the reason for this was the rate limitation on the server side. 
Let it propagate - that would be answer?